### ISSUE TITLE
AllPairs (or something similar)

### ISSUE THREAD

**Issue by MatthewDaggitt** [agda/agda-stdlib#147]
I've personally found the following predicate over lists very useful:
```agda
data AllPairs {a ℓ} {A : Set a} (_~_ : Rel A ℓ) : List A → Set (a ⊔ ℓ) where
    []  : AllPairs _~_ []
    _∷_ : ∀ {x xs} → All (x ~_) xs → AllPairs _~_ xs → AllPairs _~_ (x ∷ xs)
```
Using it you can define such useful things as:
```agda
Sorted : List A → Set _
Sorted xs = AllPairs _≤_ xs

Unique : List A → Set _
Unique xs = AllPairs _≉_ xs
```
Before I add it though, I was wondering if anyone had any opinions on:

1. The name. `AllPairs` suggests that every pair of elements in the list is related. However when the relation `_~_` is not symmetric then it only guarantees that forward pairs are related (e.g. in `a :: b :: []` it says that `a ~ b` but not that `b ~ a`). Can anyone think of a snappier or more descriptive name?

2. Its location. Should this go in `Data.List.All` or in its own file? I'm leaning towards it's own file as the `Data.List.All.Properties` file will get pretty messy.

**Comment by gallais** [comment]
@bwesterb suggests `AllSubPairs` where `Sub` is a reference to the "sub" in "sublists".

**Comment by bwesterb** [comment]
AllSubPairs or simply Ascending.

**Comment by andreasabel** [comment]
Cannot this decomposed into a function computing a list of pairs and an application of All?
```agda
triangle : List A -> List A
triangle [] = []
triangle (x :: xs) = map (x ,_) xs ++ triangle xs
```

**Comment by gallais** [comment]
This seems relevant: https://agda.github.io/agda-stdlib/Data.Vec.html#allPairs

**Comment by MatthewDaggitt** [comment]
Closed with 6965180
